src/raft.rs

```diff
@@ -1181,7 +1195,10 @@ impl<T: Storage> Raft<T> {
                     // ...or this is a PreVote for a future term...
                     (m.get_msg_type() == MessageType::MsgRequestPreVote && m.term > self.term);
                 // ...and we believe the candidate is up to date.
-                if can_vote && self.raft_log.is_up_to_date(m.index, m.log_term) {
+                if can_vote
+                    && self.raft_log.is_up_to_date(m.index, m.log_term)
+                    && (m.index > self.raft_log.last_index() || self.priority <= m.priority)
```

Comment:
priority should only work when `m.index == self.raft_log.last_index()`?

Reply:
It seems like this. A node with fewer logs cannot be a leader. The node with more logs should naturally be the leader.

Reply:
@dimstars would you like to update it?

Reply:
@hicqu Suggestions are welcome. I will make it better. (Or you mean this part needs to be modified? )